apps/web/pages/settings/platform/oauth-clients/create.tsx

```diff
@@ -16,6 +18,11 @@ export default function CreateOAuthClient() {
   const router = useRouter();
   const clientId = searchParams?.get("clientId") || "";
 
+  const { data: user, isLoading } = useMeQuery();
+  const { data: userBillingData } = useCheckTeamBilling(user?.organizationId);
+  const isPlatformUser = user?.organization.isPlatform;
+  const isPaidUser = userBillingData?.valid;
```

Comment:
we should put this logic in a hook that we can reuse in both create/edit and in future pages